iot_verifier/src/rewarder.rs

```diff
@@ -69,10 +111,23 @@ impl Rewarder {
         &mut self,
         scheduler: &Scheduler,
         iot_price: Decimal,
-    ) -> anyhow::Result<()> {
+    ) -> anyhow::Result<Duration> {
```

Comment:
not sure how i feel about this function taking on the additional responsibility of processing rewards _and_ computing the delay duration until the next attempt. since it's already returning a result, it seems like it would be more semantically correct to keep the function focused entirely on that and have it instead return a std::Result where the error could include any of the previous errors + "RewardsHalted" and/or "RewardsDelayed" and then compute the sleep duration outside of this function the way it currently does in main. In either case the rewards method still needs to be passed the Scheduler struct, so it _can_ compute the next duration, but not sure it _should_

Reply:
i agree the current incarnation is not ideal, i added this originally as i didnt want to deal with propagating errors at the time.  Meant to come back to it and then got used to it